File: app/src/main/java/com/example/bicyclefinder/DetailedBikeActivity.java
package com.example.bicyclefinder;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class DetailedBikeActivity extends AppCompatActivity {
    public static final String BIKE = "BIKE";

    private Bike wantedBike;

    EditText stelNummer;
    EditText type;
    EditText maerke;
    EditText farve;
    EditText sted;
    EditText dato;
    EditText missingFound;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detailed_bike);

        String tabString = "\t \t \t";

        Intent intent = getIntent();
        wantedBike = (Bike) intent.getSerializableExtra(BIKE);

        stelNummer = findViewById(R.id.detailedBikeFrameNoEditText);
        stelNummer.setEnabled(false);
        stelNummer.setText("Stelnummer: " + tabString + wantedBike.getFrameNumber());

        type = findViewById(R.id.detailedBikeKindOfBikeEditText);
        type.setEnabled(false);
        type.setText("Type: " + tabString + wantedBike.getKindOfBicycle());

        maerke = findViewById(R.id.detailedBikeBrandEditText);
        maerke.setEnabled(false);
        maerke.setText("Mærke: " + tabString + wantedBike.getBrand());

        farve = findViewById(R.id.detailedBikeColorEditText);
        farve.setEnabled(false);
        farve.setText("Farve: " + tabString + wantedBike.getColors());

        sted = findViewById(R.id.detailedBikePlaceEditText);
        sted.setEnabled(false);
        sted.setText("Sted: " + tabString + wantedBike.getPlace());

        dato = findViewById(R.id.detailedBikeDateEditText);
        dato.setEnabled(false);
        dato.setText("Dato: " + tabString + wantedBike.getDate());

        missingFound = findViewById(R.id.detailedBikeMissingFoundEditText);
        missingFound.setEnabled(false);
        missingFound.setText("Fremlyst / Efterlyst: " + tabString + wantedBike.getMissingFound());
    }

    public void GoBackToListActivityClick(View view) {
        finish();
    }
}

File: app/src/main/java/com/example/bicyclefinder/DetailedBikeActivity.kt
package com.example.bicyclefinder

import android.os.Bundle
import android.view.View
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

class DetailedBikeActivity : AppCompatActivity() {

    companion object {
        const val BIKE = "BIKE"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detailed_bike)
        val tabString = "\t \t \t"
        val intent = intent
        val wantedBike = intent.getSerializableExtra(BIKE) as Bike

        val frameNumber = findViewById<EditText>(R.id.detailedBikeFrameNoEditText)
        val type = findViewById<EditText>(R.id.detailedBikeKindOfBikeEditText)
        val brand = findViewById<EditText>(R.id.detailedBikeBrandEditText)
        val color = findViewById<EditText>(R.id.detailedBikeColorEditText)
        val placeFound = findViewById<EditText>(R.id.detailedBikePlaceEditText)
        val date = findViewById<EditText>(R.id.detailedBikeDateEditText)
        val missingFound = findViewById<EditText>(R.id.detailedBikeMissingFoundEditText)
        val name = findViewById<EditText>(R.id.detailedBikeNameEditText)
        val phoneNo = findViewById<EditText>(R.id.detailedBikePhoneNoEditText)

        frameNumber?.isEnabled = false
        frameNumber?.setText("Stelnummer: " + tabString + wantedBike!!.frameNumber)

        type?.isEnabled = false
        type?.setText("Type: " + tabString + wantedBike!!.kindOfBicycle)

        brand?.isEnabled = false
        brand?.setText("Mærke: " + tabString + wantedBike!!.brand)

        color?.isEnabled = false
        color?.setText("Farve: " + tabString + wantedBike!!.colors)

        placeFound?.isEnabled = false
        placeFound?.setText("Sted: " + tabString + wantedBike!!.place)

        date?.isEnabled = false
        date?.setText("Dato: " + tabString + wantedBike!!.date)

        missingFound?.isEnabled = false
        missingFound?.setText("Fremlyst / Efterlyst: " + tabString + wantedBike!!.missingFound)

        name?.isEnabled = false
        name?.setText("Fundet af: " + tabString + wantedBike!!.name)

        phoneNo?.isEnabled = false
        phoneNo?.setText("Telefonnummer: " + tabString + wantedBike!!.phoneNo)
    }

    fun GoBackToListActivityClick(view: View?) {
        finish()
    }
}


